pkg/scale/decode.go

```diff
@@ -540,6 +562,9 @@ func (ds *decodeState) decodeBytes(dstv reflect.Value) (err error) {
 	if err != nil {
 		return
 	}
+	if length > ds.maxLength {
+		return fmt.Errorf("could not decode length %v greater than max length %v", length, ds.maxLength)
```

Comment:
make this a sentinel error. I understand this package doesn't use sentinel errors, but that would help one more step for #2631